### PR TITLE
Implement gaussian_filter1d and gaussian_filter

### PR DESCRIPTION
**Pull request by adonath**
This PR implements `jax.scipy.ndimage.gaussian_filter1d` and `jax.scipy.ndimage.gaussian_filter`, addresses #7284. The implementation is mostly along the lines of what is done in scipy, whoever with some smaller differences:

- No support for the `output` argument
- No support for boundary modes other than `"constant"`, `"wrap"` and `"reflect"`, however they become available once they are supported by `jnp.pad`
-  Support of a `method` option to choose between `"direct"` and `"fft"` based convolutions.
 
Here you can find a notebook doing some comparisons of the accuracy and performance:
https://github.com/adonath/notebooks-public/blob/master/jax-vs-scipy-gaussian-filter.ipynb

The takeaways are:
 - The direct mode is asymptotically ~3x times slower than Scipy
 - The FFT is the same performance for the 1d case as Scipy (would need to check why this make sense...)
 - The ND-FFT option is asymptotically faster than Scipy, which makes sense, as Scipy relies on direct convolution (`correlate1d` I think...)

ToDo:
- [ ] Implement tests
- [ ] Investigate difference in the "`reflect`" mode

### CODE REVIEW

**Comment by jakevdp** [comment]
Thanks! However, in https://jax.readthedocs.io/en/latest/jep/18137-numpy-scipy-scope.html#scipy-ndimage we determined that `scipy.ndimage` is out-of-scope for JAX. I wonder if this functionality might fit in [dm-pix](https://github.com/google-deepmind/dm_pix) or another JAX-adjacent library?

**Comment by olly-writes-code** [comment]
> Thanks @jakevdp! This makes sense and the JEP is very clear on the rational behind. I guess It would make sense to close related issues such as #7284 or #3647 and with a reference to the JEP just to avoid other contributors hopping on it...
> 
> I would be happy to contribute this to `dm-pix`, however it seems the scope of `dm-pix` is really batched images with only a single channel dimension (typical for deep learning), and not the general n-dimensional approach from scipy. Not sure where the right place is though...

Agreed with @adonath here. The other issues (#7284 or #3647) should be closed, as the current state of those old issues creates a complicated paper trail. It would be nice to have a jax implementation of scipy ndimage. @adonath did you find anything in the end?

**Comment by adonath** [comment]
> Agreed with @adonath here. The other issues (https://github.com/jax-ml/jax/issues/7284 or https://github.com/jax-ml/jax/issues/3647) should be closed, as the current state of those old issues creates a complicated paper trail. It would be nice to have a jax implementation of scipy ndimage. @adonath did you find anything in the end?

No, I did not. I just used my own implementation of the Gaussian filters implemented in this PR.

**Comment by jakevdp** [comment]
> The other issues (#7284 or #3647) should be closed, as the current state of those old issues creates a complicated paper trail.

Thanks for the suggestion – I've closed both those issues with a pointer to the discussion at https://jax.readthedocs.io/en/latest/jep/18137-numpy-scipy-scope.html#scipy-ndimage

**Comment by jakevdp** [comment]
I think we can close this PR as well – sorry for the lack of clarity here!
